src/GPE.jl

```diff
@@ -160,45 +145,38 @@ Initialise the marginal log-likelihood of Gaussian process `gp`.
 function initialise_mll!(gp::GPE{X,Y,M,K,P,D}) where {X,Y,M,K,P,D}
     μ = mean(gp.mean,gp.x)
     Σ = cov(gp.kernel, gp.x, gp.data)
-    gp.cK = P.name.wrapper(Σ + (exp(2*gp.logNoise) + 1e-5)*I)
+    gp.cK = init_cK(P, Σ + (exp(2*gp.logNoise) + 1e-5)*I)
```

Comment:
I think the previous line should be inside init_cK. I'm thinking ahead to things like sparse GPs, where we might not want to compute Σ at all.

Reply:
Can you clarify a bit? Did you think of
```julia
function initialise_mll!(gp::GPE)
    gp.cK = init_cK(gp)
    y = gp.y - mean(gp.mean,gp.x)
    gp.alpha = gp.cK \ y
    # Marginal log-likelihood
    gp.mll = -(dot(y, gp.alpha) + logdet(gp.cK) + log2π * gp.nobs) / 2
    gp
end
init_cK(gp) = PDMat(Σ_default(gp))
init_cK(gp::GPE{X,Y,M,K,P,D}) where {X,Y,M,K,P<:ElasticPDMat,D} = ElasticPDMat(Σ_default(gp))
Σ_default(gp) = cov(gp.kernel, gp.x, gp.data) + (exp(2*gp.logNoise) + 1e-5)*I
```
Btw. I don't like to much the hard coded `1e-5`, but that's another issue.

Reply:
Yes, that's pretty much what I meant. I would have liked there to be a sensible default for `AbstractPDMat`, but it doesn't seem easy to do so elegantly.

I think your dislike of the `1e-5` default is widely shared. See #86.

Reply:
Looking back at #86, I think this is a good opportunity to get rid of the `1e-5` here. I'm thinking maybe it should be removed from `Σ_default` altogether, and leave it up to `init_cK` to add an epsilon. Have a look at `tolerant_PDMat` in `GP.jl`. The next question is how to make `tolerant_PDMat` more generic…